plugin/vault.hashicorp.com/v1alpha1/vaultsecrettransformer/VaultSecretTransformer.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/k8sdeps/kunstruct"
+	"sigs.k8s.io/kustomize/api/resid"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/yaml"
+)
+
+type plugin struct {
+	rmf              *resmap.Factory
+	ldr              ifc.Loader
+	secretReceiver   secretReceiver
+	types.TypeMeta   `json:",inline"`
+	types.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+}
+
+//noinspection GoUnusedGlobalVariable
+// nolint: golint
+var KustomizePlugin plugin
+
+func (p *plugin) Config(h *resmap.PluginHelpers, c []byte) error {
+	p.rmf = h.ResmapFactory()
+	p.ldr = h.Loader()
+
+	var err error
+	if p.secretReceiver, err = initVaultClient(); err != nil {
+		return fmt.Errorf("can't initialize Vault client: %s", err)
+	}
+
+	return yaml.Unmarshal(c, p)
+}
+
+func (p *plugin) Transform(m resmap.ResMap) error {
+
+	targetId := resid.NewResId(VaultSecretGvk, "")
+	for _, res := range m.Resources() {
+		if gvkEquals(targetId, resmap.GetOriginalId(res)) {
+			marshaled, err := res.MarshalJSON()
+			if err != nil {
+				log.Printf("WARN: Can't marshal an object: %s, %s: %s", res, res.GetName(), err)
+				continue
+			}
+			ts := VaultSecret{}
+			if err := json.Unmarshal(marshaled, &ts); err != nil {
+				log.Printf("WARN: Can't unmarshal a VaultSecret object: %s, %s: %s", res, res.GetName(), err)
+				continue
+			}
+
+			secretResource, err := p.buildSecretResource(ts.Name, ts.Namespace, ts.Spec.Path, ts.Spec.Type)
```

Comment:
instead of passing bare strings, just pass `ts`?

also, could it be called `vs` instead?  (`ts` makes me think timestamp)